### PR TITLE
ci: fix chromatic preview script

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

Fixing the script that posts Chromatic deployments links to the PR.

## Test plan

CI and comments in this PR

### CODE REVIEW

**Review by burmudar — APPROVED** [review]

File: dev/ci/post-chromatic.sh [hunk 1]
```diff
@@ -40,35 +40,36 @@ else
...
+  github_pr_comments_api_url="https://api.github.com/repos/${owner_and_repo}/issues/${pr_number}/comments"
 
-  pr_description=$(curl -sSf --request GET \
-    --url "${github_pr_api_url}" \
+  app_preview_comment_id=$(curl -sSf --request GET \
+    --url "${github_pr_comments_api_url}" \
     --user "apikey:${github_api_key}" \
     --header 'Accept: application/vnd.github.v3+json' \
-    --header 'Content-Type: application/json' | jq -r '.body')
+    --header 'Content-Type: application/json' | jq '.[] | select(.body | contains("## App preview")) | .id')
```
Comment (toolmantim) [inline dev/ci/post-chromatic.sh:50]:
```suggestion
    --header 'Content-Type: application/json' | jq '.[] | select(.body | contains("#### App preview")) | .id')
```

File: dev/ci/post-chromatic.sh [hunk 2]
```diff
@@ -40,35 +40,36 @@ else
...
 
-  # Assume Chromatic publish finishes after Render PR preview job
-  if [[ "${pr_description}" == *"## App preview"* ]]; then
-    echo "Updating PR #${pr_number} in ${owner_and_repo} description"
+  app_preview_comment_body=$(printf '%s\n' \
+    "## App preview" \
```
Comment (toolmantim) [inline dev/ci/post-chromatic.sh:53]:
```suggestion
    "#### App preview" \
```

**Review by toolmantim — COMMENTED** [review]
Nice, thank you!

Could we make it a little more minimal so we don't take up people's precious screen realestate?

Perhaps it could even just be a one-liner? e.g.

📖 [Storybook live preview](https://5f0f381c0e50750022dc6bf7-cnkkayumei.chromatic.com)

(also, just checking… does it add this to every PR, or just the ones that touch storybook things?)

File: dev/ci/post-chromatic.sh (hunk 1 above)
Comment (toolmantim) [reply dev/ci/post-chromatic.sh:50]: You can probably ignore this suggestion… the one-liner example above is probably nicer than the h4

**Comment by valerybugakov** [comment]
> also, just checking… does it add this to every PR, or just the ones that touch storybook things?

@toolmantim, it should add it to PRs that affect client code only. It's based on our custom logic atm, so it has false positives. After the Bazel migration, it will be precise.

File: dev/ci/post-chromatic.sh (hunk 1 above)
Comment (valerybugakov) [reply dev/ci/post-chromatic.sh:50]: https://github.com/sourcegraph/sourcegraph/pull/51397

Comment (valerybugakov) [reply dev/ci/post-chromatic.sh:50]: Yep, I love the one-liner suggestion!

**Pull request merged**